### PR TITLE
Issue #11343: Wrap assertThrows with getExpectedThrowable method

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #11343: Wrap assertThrows with getExpectedThrowable method

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java [hunk 1]
```diff
@@ -234,7 +234,7 @@ public void testUnmodifiableCollection() throws Exception {
         final Set<String> actualPackageNames = PackageNamesLoader
                 .getPackageNames(new TestUrlsClassLoader(Collections.emptyEnumeration()));
 
-        assertThrows(UnsupportedOperationException.class,
+        getExpectedThrowable(UnsupportedOperationException.class,
                 () -> actualPackageNames.add("com.puppycrawl.tools.checkstyle.checks.modifier"));
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java:238]: In issue we meant, use `get.....` to return value.
Now this code looks weird, we ignoring a get. 
Please update such cases to make variable and fo asser in next statement for message

File: config/import-control-test.xml [hunk 2]
```diff
@@ -20,13 +20,16 @@
...
   <!-- Disallow Junit assert in favor of Truth asserts in few exceptions -->
-  <allow class="org.junit.jupiter.api.Assertions.assertThrows"/>
   <allow class="org.junit.jupiter.api.Assertions.assertDoesNotThrow"/>
   <disallow pkg="org.junit.jupiter.api.Assertions"/>
 
   <!-- Reflection shouldn't be used in tests. -->
   <disallow pkg="java\.lang\.reflect\.*" regex="true" />
 
+    <!-- https://github.com/checkstyle/checkstyle/issues/11343 -->
+  <disallow pkg="org.junit.jupiter.api.Assertions.assertThrows"/>
```
Comment (romani) [inline config/import-control-test.xml:30]: Align xml.
Update comment to `see reason at ...`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java (hunk 1 above)
Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java:238]: I will update by this weekend.

File: config/import-control-test.xml (hunk 2 above)
Comment (Kevin222004) [reply config/import-control-test.xml:30]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java (hunk 1 above)
Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/PackageNamesLoaderTest.java:238]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java [hunk 3]
```diff
@@ -307,9 +310,10 @@ public void testIoExceptionWhenLoadingHeader() {
                 LineNumberReader.class, (mock, context) -> {
                     when(mock.readLine()).thenThrow(IOException.class);
                 })) {
-            final IllegalArgumentException ex = assertThrows(IllegalArgumentException.class, () -> {
-                check.setHeader("header");
-            });
+            final IllegalArgumentException ex =
+                    getExpectedThrowable(IllegalArgumentException.class, () -> {
+                        check.setHeader("header");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java:315]: Please add `()` to this line

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java:315]: Do same for others cases like this

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java (hunk 3 above)
Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
@Kevin222004 it would be great to follow up with https://github.com/checkstyle/checkstyle/issues/11315

**Pull request merged**